docs/EMBEDDING.md

```diff
@@ -81,3 +81,42 @@ load times. To do this please follow these steps:
 
 This will produce a `build/` directory that includes everything needed to run
 the noVNC application.
+
+## Browser Cache Issue
+
+If you serve noVNC files using a web server that provides an ETag header, and
+include any options in the query string, a nasty browser cache issue can bite
+you on upgrade, resulting in a red error box. The issue is caused by a mismatch
+between the new vnc.html (which is reloaded because the user has used it with
+new query string after the upgrade) and the old javascript files (that the
+browser reuses from its cache). The industry-standard solution to this kind of
+problems is a build step that replaces all links to all javascript files to
+include some token that is derived from the content of the said files. With
+this approach, the browser is free to cache the javascript files forever,
+because, when they change, they get a new URL that is guaranteed not to be in
+the browser cache. Unfortunately, this approach is not implemented in noVNC.
+
+The issue is tracked [on GitHub](https://github.com/novnc/noVNC/issues/1097).
+
+Until somebody submits a patch that implements this approach, you have to
+configure your web server to provide headers that prohibit browsers from
+caching noVNC files without revalidation. Example configuration snippets are
+provided below. Both of them result in a `Cache-Control: max-age=0` header.
```

Comment:
`max-age` does not guarantee the semantics we want. You need to either also have `must-validate`, or just replace it with `no-cache` which at that point is equivalent.